test/encapsulate_op/encapsulate_op_test.cc

```diff
@@ -0,0 +1,192 @@
+#include "ngraph_bridge/ngraph_utils.h"
+#include "test/test_utilities.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+namespace testing {
+#define ASSERT_OK(x) ASSERT_EQ((x), ::tensorflow::Status::OK());
+#define ASSERT_NOT_OK(x) ASSERT_NE((x), ::tensorflow::Status::OK());
+
+// Test: Create tensorflow input tensors and Compute Signature
+TEST(EncapsulateOp, ComputeSignature) {
+  NGraphEncapsulateImpl ng_encap_impl;
+
+  std::vector<tensorflow::TensorShape> input_shapes;
+  std::vector<tensorflow::Tensor> input_tensors;
+  std::vector<const Tensor*> static_input_map;
+  input_shapes.push_back({0});
+  input_shapes.push_back({2});
+  input_shapes.push_back({6, 10});
+  input_shapes.push_back({10, 10, 10});
+
+  // Create tensorflow tensors
+  for (auto const& shapes : input_shapes) {
+    Tensor input_data(DT_FLOAT, TensorShape(shapes));
+    AssignInputValuesRandom<float>(input_data, -10.0, 20.0f);
+    input_tensors.push_back(input_data);
+  }
+  int size = 4;
+  ng_encap_impl.ResizeStatic(size);
+
+  for (int i = 0; i < size; i++) {
+    ng_encap_impl.SetStatic(i, false);
+  }
+
+  for (int i = 0; i < input_tensors.size(); i++) {
+    const Tensor& input_tensor = input_tensors[i];
+    if (ng_encap_impl.GetStatic()[i]) {
+      static_input_map[i] = &input_tensor;
+    }
+  }
+  std::stringstream signature_ss;
+  ASSERT_OK(ng_encap_impl.ComputeSignature(input_tensors, input_shapes,
+                                           static_input_map, signature_ss));
+  ASSERT_EQ(signature_ss.str(), "0,;2,;6,10,;10,10,10,;/");
+}
+
+// Test: Create backend and get ngraph executable
+TEST(EncapsulateOp, GetNgExecutable) {
+  NGraphEncapsulateImpl ng_encap_impl;
+
+  std::vector<tensorflow::TensorShape> input_shapes;
+  std::vector<tensorflow::Tensor> input_tensors;
+  std::vector<const Tensor*> static_input_map;
+  input_shapes.push_back({0});
+  input_shapes.push_back({2});
+  input_shapes.push_back({6, 10});
+  input_shapes.push_back({10, 10, 10});
+
+  for (auto const& shapes : input_shapes) {
+    Tensor input_data(DT_FLOAT, TensorShape(shapes));
+    AssignInputValuesRandom<float>(input_data, -10.0, 20.0f);
+    input_tensors.push_back(input_data);
+  }
+  int size = 4;
+  ng_encap_impl.ResizeStatic(size);
+
+  for (int i = 0; i < size; i++) {
+    ng_encap_impl.SetStatic(i, false);
+  }
+
+  for (int i = 0; i < input_tensors.size(); i++) {
+    const Tensor& input_tensor = input_tensors[i];
+    if (ng_encap_impl.GetStatic()[i]) {
+      static_input_map[i] = &input_tensor;
+    }
+  }
+
+  ng_encap_impl.SetOpBackend("CPU");
+  BackendManager::CreateBackend(ng_encap_impl.GetOpBackend());
+  ng::runtime::Backend* op_backend;
+  op_backend = BackendManager::GetBackend(ng_encap_impl.GetOpBackend());
+
+  std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+
+  ASSERT_OK(ng_encap_impl.GetNgExecutable(
+      input_tensors, input_shapes, static_input_map, op_backend, ng_exec));
```

Comment:
BackendManager::ReleaseBackend() needs to be called, as clean up to free up the resources. All CreateBackend calls need to be followed by this, once we are ready to free the backend